src/NHibernate/Transform/AliasToBeanResultTransformer.cs

```diff
@@ -29,37 +31,46 @@ namespace NHibernate.Transform
 	/// sorts them by inheritance depth then by visibility from public to private, and takes those ranking first.
 	/// </remarks>
 	[Serializable]
-	public class AliasToBeanResultTransformer : AliasedTupleSubsetResultTransformer
+	public class AliasToBeanResultTransformer : AliasedTupleSubsetResultTransformer, IEquatable<AliasToBeanResultTransformer>
```

Comment:
I will add a serialization/deserialization test. Your link on SO seems unrelated: the trouble was attempting to use the map during the deserialization process. The OP states that the map was correclty filled after, but he needed it during.